avr-hal-generic/src/usart_spi.rs

```diff
@@ -0,0 +1,95 @@
+    spi::Spi<H, USART, SCLKPIN, MOSIPIN, MISOPIN, CSPIN>;
+
+// Impliment SpiOps trait for USART
+#[macro_export]
+macro_rules! add_usart_spi {
+    (
+        hal: $HAL:ty,
+        peripheral: $USART_SPI:ty,
+        register_suffix: $n:expr,
+        sclk: $sclkpin:ty,
+        mosi: $mosipin:ty,
+        miso: $misopin:ty,
+        cs: $cspin:ty,
+    ) => {
+        $crate::paste::paste! {
+            // This is quite a messy way to get the doc string working properly... but it works!
+            #[doc = concat!("**Clock:** `", stringify!($sclkpin), "`<br>**MOSI:** `", stringify!($mosipin), "`<br> **MISO:** `", stringify!($misopin), "`<br> **CS:** `", stringify!($cspin), "`")]
+            pub type [<Usart $n Spi>] = avr_hal_generic::usart_spi::UsartSpi<$HAL, $USART_SPI, $sclkpin, $mosipin, $misopin, $cspin>;
+
+            impl $crate::spi::SpiOps<$HAL, $sclkpin, $mosipin, $misopin, $cspin> for $USART_SPI {
+                fn raw_setup(&mut self, settings: &crate::spi::Settings) {
+                    use $crate::hal::spi;
+
+                    // Setup control registers
+                    // We start by setting the UBBRn to 0
+                    self.[<ubrr $n>].write(|w| unsafe {w.bits(0)});
+
+                    // We have to translate the character size register into the 2 bits which are the MSB/LSB and the phase
+                    // 5 Bit Char = MSB and 1st
+                    // 6 Bit Char = MSB and 2nd
+                    // 7 Bit Char = LSB and 1st
+                    // 8 Bit Char = LSB and 2nd
+                    self.[<ucsr $n c>].write(|w| {
+                        w.[<umsel $n>]().spi_master();
+
+                        match settings.data_order {
+                            crate::spi::DataOrder::MostSignificantFirst => match settings.mode.phase {
+                                spi::Phase::CaptureOnFirstTransition => w.[<ucsz $n>]().chr5(),
+                                spi::Phase::CaptureOnSecondTransition => w.[<ucsz $n>]().chr6(),
+                            },
+                            crate::spi::DataOrder::LeastSignificantFirst => match settings.mode.phase {
+                                spi::Phase::CaptureOnFirstTransition => w.[<ucsz $n>]().chr7(),
+                                spi::Phase::CaptureOnSecondTransition => w.[<ucsz $n>]().chr8(),
+                            },
+                        };
+
+                        match settings.mode.polarity {
+                            spi::Polarity::IdleLow => w.[<ucpol $n>]().clear_bit(),
+                            spi::Polarity::IdleHigh => w.[<ucpol $n>]().set_bit(),
+                        }
+                    });
+
+                    // Enable receiver and transmitter, and also the rec interrupt.
+                    self.[<ucsr $n b>].write(|w| w
+                        .[<txen $n>]().set_bit()
+                        .[<rxen $n>]().set_bit()
+                        .[<rxcie $n>]().set_bit()
+                    );
```

Comment:
Don't enable the interrupt by default - it should always be explicitly enabled by the user through a method call, see e.g.

https://github.com/Rahix/avr-hal/blob/65b304ec974a13f27ebc2ced21aa462c4b81ce82/avr-hal-generic/src/usart.rs#L300-L308

You don't need to implement this here, though &mdash; it should be part of the `spi` module.

Reply:
Could you elaborate on what you mean by this?

I understand why to remove it (idk why it was there to begin with), but do you want me to add methods to change the interrupts? Or something other, because I don't understand "it should be part of the spi module"